spring-amqp/src/main/java/org/springframework/amqp/support/converter/ContentTypeDelegatingMessageConverter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.amqp.support.converter;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageProperties;
+
+ /**
+  * A composite {@link MessageConverter} that delegates to an actual {@link MessageConverter}
+  * based on the contentType header. Supports a default converter when no content type matches.
+  * Note: the {@link MessageProperties} requires a content type header to select a converter
+  * when used for outbound conversion, but the converter will (generally) override it to match
+  * the actual conversion.
+  *
+  * @author Eric Rizzo
+  * @author Gary Russell
+  * @since 1.4.2
+  *
+  */
+public class ContentTypeDelegatingMessageConverter implements MessageConverter {
+
+	private final Map<String, MessageConverter> delegates = new HashMap<String, MessageConverter>();
+
+	private final MessageConverter defaultConverter;
+
+
+	/**
+	 * Constructs an instance using a default {@link SimpleMessageConverter}.
+	 */
+	public ContentTypeDelegatingMessageConverter() {
+		this(new SimpleMessageConverter());
+	}
+
+	/**
+	 * Constructs an instance using a the supplied default converter (may be null).
+	 * @param defaultConverter the converter.
+	 */
+	public ContentTypeDelegatingMessageConverter(MessageConverter defaultConverter) {
+		this.defaultConverter = defaultConverter;
```

Comment:
Just to be sure that it is expected. 
Since we don't have here `Assert.notNull`, it means that we can configure it without the `defaultConverter` to make this `ContentTypeDelegatingMessageConverter` as strict one. And this logic I see in the `getConverterForContentType`. 
Is that what you have in mind?